project/src/main/scala/edu/gemini/osgi/tools/app/RPMDistHandler.scala

```diff
@@ -82,17 +79,70 @@ class RPMDistHandler(jre: Option[String]) extends DistHandler {
     val ret = proc.waitFor
     if (ret != 0) sys.error("tar returned " + ret)
 
-    // Now create the RPM
+    // The RPM spec file.
     val specPath = new File(wd, "rpm.spec")
-    val rpmArgs = Array("rpmbuild", "-bb", s"${specPath.getAbsolutePath}")
-    val result = Runtime.getRuntime.exec(rpmArgs).waitFor()
-    if (result != 0) {
-      println("*** " + rpmArgs.mkString(" "))
-      println("*** rpmbuild returned " + result)
+
+    // Set up an SSH connection to the Linux VM where rpmbuild will be run.
+    if (config.remoteBuildInfo.isEmpty)
+      log.error("no remote build info: update OcsCredentials?")
+    val remoteBuildInfo = config.remoteBuildInfo.get
+
+    log.info(s"establishing ssh session to ${remoteBuildInfo.hostname}")
+    JSch.setConfig("StrictHostKeyChecking", "no")
+    val session = new JSch().getSession(remoteBuildInfo.username, remoteBuildInfo.hostname, remoteBuildInfo.port)
+    session.setPassword(remoteBuildInfo.password)
+    session.setTimeout(remoteBuildInfo.timeout)
+    session.connect
+
+    // Create an SFTP session to transfer to / from Linux VM.
+    val rpmBuildDir = "rpmbuild"
+    log.info("opening sftp channel")
+    val sftp = session.openChannel("sftp").asInstanceOf[ChannelSftp]
+    sftp.connect(remoteBuildInfo.timeout)
+
+    // ChannelSftp.mkdir fails with an SftpException if a directory already exists: handle this by allowing this exception.
+    def rmkdir(dir: String): Unit =
+      try {
+	sftp.mkdir(dir)
+      } catch {
+	case e: SftpException =>
+      }
```

Comment:
Unfortunately, the documentation is extremely unhelpful in this regard. The exception comes back with an `id` of `4`, which, according to the documentation, is an `SSH_FX_FAILURE`, which is used in the case of a "general failure." I suppose we could test for this specifically, but the only other two cases that come up in an `SftpChannel` operation are `SSH_FX_NO_SUCH_FILE` and `SSH_FX_OP_UNSUPPORTED`, which both seem like they wouldn't come up in an `mkdir` operation, so I'm not sure how useful that would be.

As for your other suggestion, absolutely. That's much clearer, I think.